cmake 4.0.3 flex call does not work

On Juwels Booster using CMake 4.0.3 the flex call fails:

src/lib/CMakeLists.txt

```
# cmake 4.0 uses a different syntax for the option    
if(CMAKE_MAJOR_VERSION LESS 4)    
  flex_target(tmlqcd_input_read read_input.l ${CMAKE_BINARY_DIR}/read_input.c    
              COMPILE_FLAGS "-Ca -Ptmlqcd -i")                                                                                    
else()    
  flex_target(tmlqcd_input_read read_input.l ${CMAKE_BINARY_DIR}/read_input.c    
              OPTIONS "-Ca -Ptmlqcd -i")    
endif()    
```

produces the command

```
 $ VERBOSE=1 make
[....]
[  0%] [FLEX][tmlqcd_input_read] Building scanner with flex 2.6.4
[....]
cd /p/home/jusers/kostrzewa2/juwels/build/juwelsbooster/stage_2026/gcc_14_3_openmpi_5_0_8_cuda_13/tmLQCD/tmLQCD-master-current-quda-prefetch2-529a37-sanitize/src/lib && /usr/bin/flex "-Ca -Ptmlqcd -i" -o/p/home/jusers/kostrzewa2/juwels/build/juwelsbooster/stage_2026/gcc_14_3_openmpi_5_0_8_cuda_13/tmLQCD/tmLQCD-master-current-quda-prefetch2-529a37-sanitize/read_input.c /p/home/jusers/kostrzewa2/juwels/code/tmLQCD/tmLQCD-master-current/src/lib/read_input.l
flex: unknown -C option ' '
make[2]: *** [src/lib/CMakeFiles/tmlqcd.dir/build.make:75: read_input.c] Error 1
make[2]: Leaving directory '/p/project1/isdlqcd/kostrzewa2/build/juwelsbooster/stage_2026/gcc_14_3_openmpi_5_0_8_cuda_13/tmLQCD/tmLQCD-master-current-quda-prefetch2-529a37-sanitize'
make[1]: *** [CMakeFiles/Makefile2:140: src/lib/CMakeFiles/tmlqcd.dir/all] Error 2
make[1]: Leaving directory '/p/project1/isdlqcd/kostrzewa2/build/juwelsbooster/stage_2026/gcc_14_3_openmpi_5_0_8_cuda_13/tmLQCD/tmLQCD-master-current-quda-prefetch2-529a37-sanitize'
```

I believe that `OPTIONS` must be passed as a semi-colon separated list (unlike `COMPILE_OPTIONS` in CMake < 4)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -355,9 +355,9 @@ if(CMAKE_BUILD_TYPE STREQUAL "SANITIZE")
   if(CMAKE_CXX_COMPILER_ID MATCHES "GNU|Clang")
     message(STATUS "Building with AddressSanitizer")
     # Apply AddressSanitizer flags for the SANITIZE build type
-
-    add_compile_options($<$<COMPILE_LANGUAGE:CXX,C>:-fsanitize=address
-                        -fno-omit-frame-pointer -g>)
+    add_compile_options($<$<COMPILE_LANGUAGE:CXX,C>:-fsanitize=address>) 
+    add_compile_options($<$<COMPILE_LANGUAGE:CXX,C>:-fno-omit-frame-pointer>)
+    add_compile_options($<$<COMPILE_LANGUAGE:CXX,C>:-g>)
     add_link_options(-fsanitize=address)
   else()
     message(WARNING "AddressSanitizer is not supported for this compiler")

--- a/src/lib/CMakeLists.txt
+++ b/src/lib/CMakeLists.txt
@@ -386,7 +386,7 @@ if(CMAKE_MAJOR_VERSION LESS 4)
               COMPILE_FLAGS "-Ca -Ptmlqcd -i")
 else()
   flex_target(tmlqcd_input_read read_input.l ${CMAKE_BINARY_DIR}/read_input.c
-              OPTIONS "-Ca -Ptmlqcd -i")
+              OPTIONS "-Ca;-Ptmlqcd;-i")
 endif()
 
 # create a target library with namespacing because cmake does not know name

